include/s3select_functions.h

```diff
@@ -530,34 +612,58 @@ struct _fn_to_timestamp : public base_function
 
   bool datetime_validation()
   {
-    //TODO temporary , should check for leap year
-
-    if(yr<1700 || yr>2050)
-    {
-      return false;
-    }
-    if (mo<1 || mo>12)
-    {
-      return false;
-    }
-    if (dy<1 || dy>31)
-    {
-      return false;
-    }
-    if (hr>23)
-    {
-      return false;
-    }
-    if (dy>59)
-    {
-      return false;
-    }
-    if (sc>59)
+    if (yr>=1400 && yr<=9999 && mo>=1 && mo<=12 && dy>=1 && hr<24 && mn<60 && sc<60 && tz_hr<24 && tz_mn<60)
     {
-      return false;
+      switch (mo)
+      {
+        case 1:
+        case 3:
+        case 5:
+        case 7:
+        case 8:
+        case 10:
+        case 12:
+		if(dy<=31)
+		{
+                  return true;
+                }
+                break;
+        case 4:
+        case 6:
+        case 9:
+        case 11:
+                if(dy<=30)
+                {
+                  return true;
+                }
+                break;
+        case 2:
+                if(dy<=29)
```

Comment:
it branch into leap year conditions for all the days of Feb,
it's better to "reverse" it, i.e. if(dy>=28)

Reply:
Done